src/libostree/ostree-repo-pull.c

```diff
@@ -1286,9 +1336,14 @@ meta_fetch_on_complete (GObject           *object,
   g_assert (pull_data->n_outstanding_metadata_fetches > 0);
   pull_data->n_outstanding_metadata_fetches--;
   pull_data->n_fetched_metadata++;
-  check_outstanding_requests_handle_error (pull_data, &local_error);
+
+  if (should_retry_request (local_error, fetch_data->n_retries_remaining--))
+    enqueue_one_object_request_s (pull_data, g_steal_pointer (&fetch_data));
+  else
+    check_outstanding_requests_handle_error (pull_data, &local_error);
+
   if (free_fetch_data)
```

Comment:
Not new here, though I wonder why we need these bools instead of using the `g_steal_pointer` pattern higher up. Anyway, we can look into that later!

Reply:
I’m not sure it applies in this particular instance, but there are other similar patterns when enqueuing other structs, where `g_steal_pointer (&the_struct)` can’t be used because `the_struct` needs to be referred to again lower down, before it’s eventually potentially freed.

In any case, the code could be restructured to keep a `const` pointer to whatever it needs to refer to later, or something.